src/main/java/com/epam/aidial/cfg/domain/service/ModelService.java

```diff
@@ -77,6 +80,22 @@ public void updateModel(String modelName, Model model) {
                 .orElseThrow(() -> new RuntimeException("Unable to update model " + model.getDeployment().getName()));
     }
 
+    @Transactional
+    public String updateModel(String modelName, Model model, String hash) {
+        if (model == null) {
+            throw new RuntimeException("Unable to update model " + modelName);
```

Comment:
try to use more specific exception and more specific messages, e.g. IllegalArgumentException("Unable to update model '" + modelName + "', "model does not provided")